spec/concepts/api/v1/users/registrations/contract/create_spec.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+RSpec.describe Api::V1::Users::Registrations::Contract::Create, type: :contract do
+  subject(:contract) do
+    described_class
+  end
+
+  let(:result) { contract.call(params) }
+  let(:i18n_path) { %i[users registrations contract create] }
+
+  describe 'Success' do
+    context 'when valid params' do
+      let(:user) { build(:user) }
+      let(:params) { { email: user.email, password: user.password, password_confirmation: user.password } }
+
+      it 'returns valid object' do
+        expect(result).to be_success
+        expect(result.errors).to be_empty
+      end
+    end
+  end
+
+  describe 'Failure' do
+    describe 'With invalid credentials' do
+      context 'when missed required fields' do
+        let(:params) { {} }
+
+        it 'return object with errors' do
+          expect(result).to be_failure
+          expect_errors(result, %i[email password password_confirmation], 'is missing')
+        end
+      end
+
+      context 'when required fields are blank' do
+        let(:params) { { email: '', password: '', password_confirmation: '' } }
+
+        it 'return object with errors' do
```

Comment:
all of these can be definded and used as [shared examples](https://relishapp.com/rspec/rspec-core/v/3-10/docs/example-groups/shared-examples):

```ruby
shared_examples 'returns failed contract with errors' do
  it 'returns failed contract with errors' do
    expect_errors(result, attrs, error_message)
    expect(result).to be_failure
  end
end
```

Reply:
Check please, I tried.